contracts/test/integration/vip/vip15.sol

```diff
@@ -0,0 +1,477 @@
+                arguments: abi.encodeWithSignature(
+                    "revokeRole(bytes32,address)",
+                    CANCELLER_ROLE,
+                    MainnetAddresses.EOA_4
+                ),
+                description: "Revoke canceller role from EOA 4"
+            })
+        );
+
+        /// ------- role granted in timelock -------
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.TIMELOCK_CONTROLLER,
+                arguments: abi.encodeWithSignature(
+                    "grantRole(bytes32,address)",
+                    PROPOSER_ROLE,
+                    MainnetAddresses.GOVERNOR
+                ),
+                description: "Grant proposer role to multisig"
+            })
+        );
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.TIMELOCK_CONTROLLER,
+                arguments: abi.encodeWithSignature(
+                    "grantRole(bytes32,address)",
+                    CANCELLER_ROLE,
+                    MainnetAddresses.GOVERNOR
+                ),
+                description: "Grant canceller role to multisig"
+            })
+        );
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.TIMELOCK_CONTROLLER,
+                arguments: abi.encodeWithSignature(
+                    "grantRole(bytes32,address)",
+                    EXECUTOR_ROLE,
+                    address(0)
+                ),
+                description: "Allow execution by any address"
+            })
+        );
+
+        /// ------- disconnect psms -------
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.ERC20ALLOCATOR,
+                arguments: abi.encodeWithSignature(
+                    "disconnectPSM(address)",
+                    MainnetAddresses.VOLT_USDC_PSM
+                ),
+                description: "Disconnect old USDC PSM from the ERC20 Allocator"
+            })
+        );
+
+        proposal.push(
+            ITimelockSimulation.action({
+                value: 0,
+                target: MainnetAddresses.ERC20ALLOCATOR,
+                arguments: abi.encodeWithSignature(
+                    "disconnectPSM(address)",
+                    MainnetAddresses.VOLT_DAI_PSM
+                ),
+                description: "Disconnect old DAI PSM from the ERC20 Allocator"
+            })
+        );
+    }
+
+    function getMainnetProposal()
+        public
+        view
+        override
+        returns (ITimelockSimulation.action[] memory prop)
+    {
+        prop = proposal;
+    }
+
+    function mainnetSetup() public override {}
+
+    function mainnetValidate() public override {
+        assertTrue(!core.isPCVController(address(allocator)));
```

Comment:
add additional assertions here that all state transitions that we wanted to happen happened.